### PR TITLE
Allow site admin to disable mirrors

### PR DESCRIPTION
**Pull request by jolheiser**
Wording is consistent with `FORCE_PRIVATE`

mirrors

This will stop the creation of *new* mirror migrations. For existing mirrors, the site admin will need to turn off the cron for updating them (or delete them).

### CODE REVIEW

**Review by CirnoT — APPROVED** [review]

File: custom/conf/app.ini.sample [hunk 1]
```diff
@@ -50,6 +50,8 @@ DISABLED_REPO_UNITS =
 DEFAULT_REPO_UNITS = repo.code,repo.releases,repo.issues,repo.pulls,repo.wiki
 ; Prefix archive files by placing them in a directory named after the repository
 PREFIX_ARCHIVE_FILES = true
+; Disable new mirror migrations
```
Comment (guillep2k) [inline custom/conf/app.ini.sample]:
```suggestion
; Disable the creation of mirrors. Pre-existing mirrors remain valid.
```

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 2]
```diff
@@ -69,6 +69,7 @@ Values containing `#` or `;` must be quoted using `` ` `` or `"""`.
 - `ENABLE_PUSH_CREATE_USER`:  **false**: Allow users to push local repositories to Gitea and have them automatically created for a user.
 - `ENABLE_PUSH_CREATE_ORG`:  **false**: Allow users to push local repositories to Gitea and have them automatically created for an org.
 - `PREFIX_ARCHIVE_FILES`: **true**: Prefix archive files by placing them in a directory named after the repository.
+- `DISABLE_MIRRORS`: **false**: Disable *new* mirror migrations
```
Comment (guillep2k) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md]:
```suggestion
- `DISABLE_MIRRORS`: **false**: Disable the creation of **new** mirrors. Pre-existing mirrors remain valid.
```

**Review by guillep2k — APPROVED** [review]
Just a comment change suggestion. Otherwise LG-TM.

File: custom/conf/app.ini.sample (hunk 1 above)
Comment (jolheiser) [reply custom/conf/app.ini.sample]:
```suggestion
; Disable the creation of new mirrors. Pre-existing mirrors remain valid.
```

**Comment by techknowlogick** [comment]
ping LG-TM

**Comment by 6543** [comment]
@Codeberg-org this can be usefull for you :)

File: routers/api/v1/repo/migrate.go [hunk 3]
```diff
@@ -118,7 +118,7 @@ func Migrate(ctx *context.APIContext, form auth.MigrateRepoForm) {
 		RepoName:       form.RepoName,
 		Description:    form.Description,
 		Private:        form.Private || setting.Repository.ForcePrivate,
-		Mirror:         form.Mirror,
+		Mirror:         form.Mirror && !setting.Repository.DisableMirrors,
```
Comment (6543) [inline routers/api/v1/repo/migrate.go:121]: I would had prefered to return an error

Comment (6543) [reply routers/api/v1/repo/migrate.go:121]: ->  #11757

**Pull request merged**